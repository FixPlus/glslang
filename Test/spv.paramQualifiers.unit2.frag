#version 460 core

float foo(in float, out int, inout vec3);

out float cout;

void main()
{
    int x;
    vec3 v = vec3(0.f, 1.f, 2.f);
    float r = foo(1.f, x, v);
}
