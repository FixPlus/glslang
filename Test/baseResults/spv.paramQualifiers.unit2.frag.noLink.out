spv.paramQualifiers.unit2.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:7  Function Definition: main( ( global void)
0:7    Function Parameters: 
0:?     Sequence
0:10      Sequence
0:10        move second child to first child ( temp highp 3-component vector of float)
0:10          'v' ( temp highp 3-component vector of float)
0:10          Constant:
0:10            0.000000
0:10            1.000000
0:10            2.000000
0:11      Sequence
0:11        move second child to first child ( temp highp float)
0:11          'r' ( temp highp float)
0:11          Function Call: foo(f1;i1;vf3; ( global highp float)
0:11            Constant:
0:11              1.000000
0:11            'x' ( temp highp int)
0:11            'v' ( temp highp 3-component vector of float)
0:?   Linker Objects
0:?     Linker Function: foo(f1;i1;vf3; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( out highp int)
0:?         '' ( inout highp 3-component vector of float)
0:?     'cout' ( out highp float)

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 33

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 32
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 460
                              Name 4  "main"
                              Name 16  "foo(f1;i1;vf3;"
                              Name 17  "v"
                              Name 22  "r"
                              Name 23  "x"
                              Name 24  "param"
                              Name 25  "param"
                              Name 26  "param"
                              Name 32  "cout"
                              Decorate 16(foo(f1;i1;vf3;) Linkage Attributes 678391654 1765486950 1719024433 15155 1
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypePointer Function 6(float)
               8:             TypeInt 32 1
               9:             TypePointer Function 8(int)
              10:             TypeVector 6(float) 3
              11:             TypePointer Function 10(fvec3)
              12:             TypeFunction 6(float) 7(ptr) 9(ptr) 11(ptr)
              18:    6(float) Constant 0
              19:    6(float) Constant 1065353216
              20:    6(float) Constant 1073741824
              21:   10(fvec3) ConstantComposite 18 19 20
              31:             TypePointer Output 6(float)
        32(cout):     31(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
           17(v):     11(ptr) Variable Function
           22(r):      7(ptr) Variable Function
           23(x):      9(ptr) Variable Function
       24(param):      7(ptr) Variable Function
       25(param):      9(ptr) Variable Function
       26(param):     11(ptr) Variable Function
                              Store 17(v) 21
                              Store 24(param) 19
              27:   10(fvec3) Load 17(v)
                              Store 26(param) 27
              28:    6(float) FunctionCall 16(foo(f1;i1;vf3;) 24(param) 25(param) 26(param)
              29:      8(int) Load 25(param)
                              Store 23(x) 29
              30:   10(fvec3) Load 26(param)
                              Store 17(v) 30
                              Store 22(r) 28
                              Return
                              FunctionEnd
16(foo(f1;i1;vf3;):    6(float) Function None 12
              13:      7(ptr) FunctionParameter
              14:      9(ptr) FunctionParameter
              15:     11(ptr) FunctionParameter
                              FunctionEnd
